torchdynamo/_eval_frame.c

```diff
@@ -238,10 +283,11 @@ inline static PyObject *eval_custom_code(PyThreadState *tstate,
   DEBUG_CHECK(nlocals_new >= nlocals_old);
 
   PyFrameObject *shadow =
-      _PyFrame_New_NoTrack(tstate, code, frame->f_globals, NULL);
+      PyFrame_New(tstate, code, frame->f_globals, NULL);
   if (shadow == NULL) {
     return NULL;
   }
+  PyObject_GC_UnTrack(shadow);
```

Comment:
I think at the end of this function we call `PyObject_GC_Track(shadow)`, so I think this was an optimization I copied from `cpython/.../call.c` to avoid ever starting to track the frame.  Given this patch removes that optimization, perhaps we can simplify things.

I'd suggest removing this line, then replacing L308 to L315 with just `Py_DECREF(shadow)`.